gossip/components/processor.c

```diff
@@ -7,269 +7,384 @@
+                              NUM_TRITS_SERIALIZED_TRANSACTION) !=
+        NUM_TRITS_SERIALIZED_TRANSACTION) {
+      log_warning(PROCESSOR_LOGGER_ID, "Invalid transaction bytes\n");
+      ret = RC_PROCESSOR_INVALID_TRANSACTION;
+      goto failure;
     }
-    // TODO(thibault): Transaction validation
-    // TODO(thibault): Add to cache
 
-    TRIT_ARRAY_DECLARE(hash, NUM_TRITS_HASH);
-    trit_array_set_trits(&hash, (*tx)->hash, NUM_TRITS_HASH);
+    // Deserializes th transaction
+    if (transaction_deserialize_from_trits(&transaction,
+                                           transaction_flex_trits) !=
+        NUM_TRITS_SERIALIZED_TRANSACTION) {
+      log_warning(PROCESSOR_LOGGER_ID, "Deserializing transaction failed\n");
+      ret = RC_PROCESSOR_INVALID_TRANSACTION;
+      goto failure;
+    }
 
+    // Validates the transaction
+    if (!iota_consensus_transaction_validate(processor->transaction_validator,
+                                             &transaction)) {
+      log_warning(PROCESSOR_LOGGER_ID, "Invalid transaction\n");
+      ret = RC_PROCESSOR_INVALID_TRANSACTION;
+      goto failure;
+    }
+
+    // TODO Add transaction hash to cache
+
+    memcpy(hash, transaction.hash, FLEX_TRIT_SIZE_243);
+
+    // Checks if the transaction is already persisted
     if ((ret = iota_tangle_transaction_exist(
-             state->tangle, TRANSACTION_FIELD_HASH, &hash, &exists))) {
-      return ret;
+             processor->tangle, TRANSACTION_FIELD_HASH, &key, &exists)) !=
+        RC_OK) {
+      log_warning(PROCESSOR_LOGGER_ID,
+                  "Checking if transaction exists failed\n");
+      goto failure;
     }
-    if (exists == false) {
-      neighbor->nbr_new_tx++;
-      // Store new transaction
-      log_debug(PROCESSOR_COMPONENT_LOGGER_ID, "Storing new transaction\n");
-      if ((ret = iota_tangle_transaction_store(state->tangle, *tx))) {
-        log_warning(PROCESSOR_COMPONENT_LOGGER_ID,
-                    "Storing new transaction failed\n");
-        neighbor->nbr_invalid_tx++;
-        return ret;
+
+    if (!exists) {
+      // Stores the new transaction
+      log_debug(PROCESSOR_LOGGER_ID, "Storing new transaction\n");
+      if ((ret = iota_tangle_transaction_store(processor->tangle,
+                                               &transaction)) != RC_OK) {
+        log_warning(PROCESSOR_LOGGER_ID, "Storing new transaction failed\n");
+        goto failure;
       }
-      // TODO(thibault): Store transaction metadata
-      // Broadcast new transaction
-      log_debug(PROCESSOR_COMPONENT_LOGGER_ID,
-                "Propagating packet to broadcaster\n");
+
+      // Checks solidity of the transaction
       if ((ret =
-               broadcaster_on_next(&state->node->broadcaster, tx_flex_trits))) {
-        log_warning(PROCESSOR_COMPONENT_LOGGER_ID,
-                    "Propagating packet to broadcaster failed\n");
+               iota_consensus_transaction_solidifier_check_and_update_solid_state(
```

Comment:
Will this also cause a traversal? If so, this is a potential DoS vector

Reply:
No, just checking solidity of current tx, branch and trunk